src/version_weaver.cpp

```diff
@@ -2,20 +2,22 @@
 #include <algorithm>
 #include <cctype>
 #include <regex>
+#include <sstream>
```

Comment:
I would avoid sstream. 

Have you considered  std::format... 

https://en.cppreference.com/w/cpp/utility/format/format.html

Reply:
this link will give me super information and I will use it to write cpp better, thank you again, I sent the update